src/librustdoc/html/render/mod.rs

```diff
@@ -1473,15 +1506,16 @@ pub(crate) fn notable_traits_button(ty: &clean::Type, cx: &Context<'_>) -> Optio
         }
     }
 
-    if has_notable_trait {
+    has_notable_trait.then(|| {
         cx.types_with_notable_traits.borrow_mut().insert(ty.clone());
-        Some(format!(
-            " <a href=\"#\" class=\"tooltip\" data-notable-ty=\"{ty}\">ⓘ</a>",
-            ty = Escape(&format!("{:#}", ty.print(cx))),
-        ))
-    } else {
-        None
-    }
+        fmt::from_fn(|f| {
+            write!(
+                f,
+                " <a href=\"#\" class=\"tooltip\" data-notable-ty=\"{ty}\">ⓘ</a>",
+                ty = Escape(&format!("{:#}", ty.print(cx))),
```

Comment:
Not at all needed in this PR, but I wounder if this temporary allocation here can also go.

Reply:
Yeah, I was wondering that too. I think it's unavoidable, since `Escape` needs to actually go over the formatted string and escape certain characters. The only way to create a "lazy" `Escape` construct would be to modify std's `Formatter`, which obviously is not an option...
But maybe I'm missing something.